sos/plugins/etcd.py

```diff
@@ -16,40 +16,38 @@
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
 from sos.plugins import Plugin, RedHatPlugin
-import socket
 
 
 class etcd(Plugin, RedHatPlugin):
 
     """etcd plugin
     """
 
-    def port_report(self):
-        """ Until etcd v2, the daemon listens on port 4001 for backward
-        compatibility of v2.0 eariler. But etcd v3 no loger litens on 
-        port 4001 and it listens on port 2379 only. 
-        To collect information in any version, this plugin should check 
-        which port is available.
-        """
-        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        try:
-            self.sock.connect(("localhost", 2379))
-        except socket.error, msg:
-            self.sock.close()
-            return "4001"
-        self.sock.close()
-        return "2379"
+    def report_url_and_port(self):
+        access_url_port = 'http://localhost:4001'
+        peer_flag = False
+        f = open('/etc/etcd/etcd.conf', 'r')
```

Comment:
You need to handle exceptions whenever you are doing file IO in a plugin; otherwise a single exception during setup or collection will lead to any remaining actions for that plugin being skipped.